tidy3d/plugins/mode/mode_solver.py

```diff
@@ -317,27 +318,410 @@ def data_raw(self) -> ModeSolverData:
-        mode_solver_data = self._data_on_yee_grid()
+        if not self.mode_spec.bend_correction:
+            # Compute data on the Yee grid
+            mode_solver_data = self._data_on_yee_grid()
 
-        # Colocate to grid boundaries if requested
-        if self.colocate:
-            mode_solver_data = self._colocate_data(mode_solver_data=mode_solver_data)
+            # Colocate to grid boundaries if requested
+            if self.colocate:
+                mode_solver_data = self._colocate_data(mode_solver_data=mode_solver_data)
 
-        # normalize modes
-        self._normalize_modes(mode_solver_data=mode_solver_data)
+            # normalize modes
+            self._normalize_modes(mode_solver_data=mode_solver_data)
 
-        # filter polarization if requested
-        if self.mode_spec.filter_pol is not None:
-            self._filter_polarization(mode_solver_data=mode_solver_data)
+            # filter polarization if requested
+            if self.mode_spec.filter_pol is not None:
+                self._filter_polarization(mode_solver_data=mode_solver_data)
 
-        # sort modes if requested
-        if self.mode_spec.track_freq and len(self.freqs) > 1:
-            mode_solver_data = mode_solver_data.overlap_sort(self.mode_spec.track_freq)
+            # sort modes if requested
+            if self.mode_spec.track_freq and len(self.freqs) > 1:
+                mode_solver_data = mode_solver_data.overlap_sort(self.mode_spec.track_freq)
 
-        self._field_decay_warning(mode_solver_data.symmetry_expanded)
+            self._field_decay_warning(mode_solver_data.symmetry_expanded)
 
-        return mode_solver_data
+            return mode_solver_data
+        else:
+            # Create a reference solver with rotated geometries for interpolations
+            rotrated_strucs = self.rotate_structure
+            rotated_sim = self.simulation.updated_copy(structures=rotrated_strucs)
+            rotated_mode_spec = self.mode_spec.updated_copy(
+                bend_correction=False, angle_theta=0, angle_phi=0
+            )
+            solver_ref = self.updated_copy(simulation=rotated_sim, mode_spec=rotated_mode_spec)
+            solver_data_ref = solver_ref.data_raw
+
+            _, id_plane = self.pop_axis((0, 1, 2), axis=self.normal_axis)
+            idx_bend = id_plane[self.mode_spec.bend_axis]  # Bend axis id
+
+            rt_bent_center = [0, 0, 0]
+            id_rotate = 3 - self.normal_axis - idx_bend
+            radius = self.mode_spec.bend_radius
+
+            rt_bent_center[self.normal_axis] = self.plane.center[self.normal_axis]
+            rt_bent_center[idx_bend] = self.plane.center[idx_bend]
+            rt_bent_center[id_rotate] = self.plane.center[id_rotate] - radius
+            cylindrical_solver_data_ref = self.process_car_2_cyn_field(
+                solver_data_ref, idx_bend, rt_bent_center
+            )
+
+            # To do: generate coords for the mode monitor
```

Comment:
What do you mean by that?

Also as a side note use `TODO:` in the comments, that's the standard keyword we use so it's good to search through the codebase for it, and also many IDEs have special handling for it, like they can show them to you nicely.